### DISCUSSION TITLE
Is IOX_SHM_WRITE_ZEROS_ON_CREATION necessary on Linux platform?

### DISCUSSION THREAD

**Discussion by wjswjq** [eclipse-iceoryx/iceoryx#2356] [General]
Hi maintainers, 
According to man page,  if we create SharedMemoryObject by shm_open(O_CREATE) & ftruncate, the shm is zero initialized by default. So is IOX_SHM_WRITE_ZEROS_ON_CREATION necessary on Linux platform?
[shm_open](https://man7.org/linux/man-pages/man3/shm_open.3.html)

> A new shared memory object initially has zero length—the
              size of the object can be set using [ftruncate(2)](https://man7.org/linux/man-pages/man2/ftruncate.2.html).  The
              newly allocated bytes of a shared memory object are
              automatically initialized to 0.

**Comment by elfenpiff** [comment]
Maybe, I do not know if this is valid for all libc implementations. If you combine iceoryx on linux with dietlibc or musl libc they may behave differently.

This option also solves another problem. With `shm_open`, `ftruncate` and `mmap` you can also over-allocate and when you then really require the memory during runtime you will end up with a `SEGFAULT`. One reliable way of ensuring that the acquired memory is really available is to zero the memory.
